src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/CPSProject_IProjectContext.cs

```diff
@@ -103,15 +127,27 @@ public void AddProjectReference(IProjectContext project, MetadataReferenceProper
 
         public void RemoveProjectReference(IProjectContext project)
         {
-            foreach (var projectReference in GetCurrentProjectReferences())
+            var referencedProject = GetAbstractProject(project);
+
+            // AbstractProject and ProjectTracker track project references using the project bin output path.
```

Comment:
I don't get this comment. Project references given to us normally we should track directly, and not require a bin path at all.